sunpy/map/mapbase.py

```diff
@@ -726,19 +726,24 @@ def max(self, *args, **kwargs):
     @staticmethod
     def _parse_fits_unit(unit_str):
         replacements = {'gauss': 'G',
-                        'dn': 'ct',
-                        'dn/s': 'ct/s',
                         'counts / pixel': 'ct/pix',}
         if unit_str.lower() in replacements:
             unit_str = replacements[unit_str.lower()]
         unit = u.Unit(unit_str, format='fits', parse_strict='silent')
         if isinstance(unit, u.UnrecognizedUnit):
-            warn_metadata(f'Could not parse unit string "{unit_str}" as a valid FITS unit.\n'
-                          f'See {_META_FIX_URL} for how to fix metadata before loading it '
-                          'with sunpy.map.Map.\n'
-                          'See https://fits.gsfc.nasa.gov/fits_standard.html for '
-                          'the FITS unit standards.')
-            unit = None
+            # NOTE: Special case DN here as it is not part of the FITS standard, but
+            # is widely used and is also a recognized astropy unit
+            unit_str_components = unit.to_string().replace('/', ' ').replace('(', '').replace(')', '')
```

Comment:
What on earth caused you to need this line? This is great and I am here for it.

Reply:
I guess I removed the comment explaining this. Because anything with DN is not FITS-compliant, the whole unit becomes an `UnrecognizedUnit` and you cannot decompose it into individual `Unit` objects. Thus, you have to parse it as a string. The `replace` methods remove non unit characters that could be present in the string. 

The goal here is to breakup the unit into components to check if one of them is DN. Just checking for "DN" in the unit string is ambiguous because it's possible some unit could include a contiguous "dn" in it somewhere. 

If there is a better way to do this, I'm all for it. I don't like this approach.

Reply:
> I guess I removed the comment explaining this. Because anything with DN is not FITS-compliant, the whole unit becomes an `UnrecognizedUnit` and you cannot decompose it into individual `Unit` objects. Thus, you have to parse it as a string. The `replace` methods remove non unit characters that could be present in the string.
> 
> The goal here is to breakup the unit into components to check if one of them is DN. Just checking for "DN" in the unit string is ambiguous because it's possible some unit could include a contiguous "dn" in it somewhere.
> 

Ah! 

> I don't like this approach.

I do.

What kind of strings do we have to handle?
If I can see some examples, we could hopefully come up with another method.

Reply:
See the updated `_parse_fits_unit` tests for some examples.

Reply:
Rather than this abomination why not just do this:
```python
unit = u.Unit(unit_str)
if u.DN not in base_unit.bases:
    warn_metadata(...)
    unit = None
```

Reply:
Does that work when you have an `UnrecognizedUnit`? I did not try that but yes that would certainly be preferable (though less fun).

Reply:
Seems to work, at least the tests pass.